app/src/main/java/org/simple/clinic/deeplink/DeepLinkUpdate.kt

```diff
@@ -18,18 +18,27 @@ class DeepLinkUpdate : Update<DeepLinkModel, DeepLinkEvent, DeepLinkEffect> {
       model: DeepLinkModel,
       event: UserFetched
   ): Next<DeepLinkModel, DeepLinkEffect> {
-    return when {
-      event.user?.loggedInStatus == User.LoggedInStatus.LOGGED_IN -> {
-        val effect = if (model.patientUuid != null) {
-          FetchPatient(model.patientUuid)
-        } else {
-          ShowNoPatientUuidError
-        }
-        dispatch(effect)
-      }
-      else -> {
-        dispatch(NavigateToSetupActivity)
-      }
+    return if (event.user?.loggedInStatus == User.LoggedInStatus.LOGGED_IN) {
+      checkIfMedicalOfficerHasCapabilitiesToTeleconsult(model, event)
+    } else {
+      dispatch(NavigateToSetupActivity)
+    }
+  }
+
+  private fun checkIfMedicalOfficerHasCapabilitiesToTeleconsult(model: DeepLinkModel, event: UserFetched): Next<DeepLinkModel, DeepLinkEffect> {
```

Comment:
Same as other comment that the name of this method is exposing a check that it is making and not what the primary goal of the method (to show the log teleconsult screen) is.